django/boss/throttling.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2019 The Johns Hopkins University Applied Physics Laboratory
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from rest_framework.exceptions import Throttled
+
+from django.utils.translation import ugettext_lazy as _
+from django.conf import settings as django_settings
+
+import json
+import bossutils
+import redis
+
+def parse_limit(val):
```

Comment:
Can you add comments to this? Assume val contains a string with a count suffixed with K, M, G, T, or P? What do those letters mean? Is it a type of call made by the user?

Reply:
Added comments to the whole file, including for format of `val` and what the suffixes are.

This method is used to decode the throttling limits that are stored in Vault and helps makes those limits human readable.